che-incubator/devworkspace-telemetry-amplitude-plugin
# Send events to Amplitude via Segment

This PR is largely based off of the code in https://github.com/che-incubator/devworkspace-telemetry-woopra-plugin

The main differences between the Woopra telemetry plugin and the amplitude telemetry plugin is:
* Config options `WOOPRA_DOMAIN` and `WOOPRA_DOMAIN_ENDPOINT` are removed
* Woopra pinging code, which was used to let Woopra know that the session is still active, has been removed
* SessionId is being added to the payload. SessionId is used to track sessions in Amplitude.

### How to test:
Like the Woopra telemetry extension, this telemetry extension can be built using JVM mode or native build mode. The readme provides instructions on how to build

To quickly build and test, build the image in JVM mode:
```
mvn package && docker build -f src/main/docker/Dockerfile.jvm -t quay.io/username/reponame:next . && docker push quay.io/username/reponame:next
```

I have also built the JVM image and pushed it to `quay.io/dkwon17/test:telemetry_session_id`

To test the telemetry extension within a running workspace:
1. Create a workspace in Eclipse Che (ex. with a sample)
2. Add the following to the devworkspace's yaml:
```
spec:
  contributions:
    - name: telemetry
      uri: https://gist.githubusercontent.com/dkwon17/91590cf91720b529b9b2e38010f69b8b/raw/12be51c6a6b791e96465eea10796189b4c89bc3e/gistfile1.txt
```
3. Restart the workspace
4. Once the editor opens, make some edits within the editor
5. Check that the events are being sent. This can be done in two ways:

<details><summary>If you have access to the Amplitude dashboard</summary>

Stop the workspace and go to the [Amplitude dashboard](https://analytics.amplitude.com/redhat/project/426592/search/amplitude_id%3D554946294455) and verify that the `Start`, `Edit`, and `Stop` events are sent, and verify that the three events have the same sessionId:
![image](https://github.com/che-incubator/devworkspace-telemetry-amplitude-plugin/assets/83611742/561ef73d-b110-44ad-b964-944fa7e5a610)
<img width="565" alt="image" src="https://github.com/che-incubator/devworkspace-telemetry-amplitude-plugin/assets/83611742/88421ddf-0588-49d7-9b9b-a836a49cbe96">

</details>


<details><summary>If you don't have access to the Amplitude dashboard</summary>

The `devworkspace-telemetry-amplitude-plugin` container from the workspace pod should have lines that contain the following two strings:
```
sending Start Workspace in Che
```

```
sending Edit Workspace File in Che
```

</details>
